File: app/src/main/java/com/example/myapplication/util/DateUtils.java
package com.example.myapplication.util;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public final class DateUtils {
    private static final String DATE_PATTERN = "EEE, MMM d yyyy 'at' h:mm a";

    private DateUtils() {
    }

    public static String epochToReadable(long epochMs) {
        SimpleDateFormat format = new SimpleDateFormat(DATE_PATTERN, Locale.getDefault());
        return format.format(new Date(epochMs));
    }

    public static long readableToEpoch(String dateString) {
        SimpleDateFormat format = new SimpleDateFormat(DATE_PATTERN, Locale.getDefault());
        try {
            Date parsedDate = format.parse(dateString);
            return parsedDate != null ? parsedDate.getTime() : 0L;
        } catch (ParseException e) {
            return 0L;
        }
    }
}

File: app/src/main/java/com/example/a4200gp/util/DateUtils.kt
package com.example.a4200gp.util

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object DateUtils {
    private const val DATE_PATTERN = "EEE, MMM d yyyy 'at' h:mm a"

    fun epochToReadable(epochMs: Long): String {
        val format = SimpleDateFormat(DATE_PATTERN, Locale.getDefault())
        return format.format(Date(epochMs))
    }

    fun readableToEpoch(dateString: String): Long {
        val format = SimpleDateFormat(DATE_PATTERN, Locale.getDefault())
        return format.parse(dateString)?.time ?: 0L
    }
}
